### DISCUSSION TITLE
Removing small (2-4) pixel false grains from within larger grains?

### DISCUSSION THREAD

**Discussion by jps0084** [mtex-toolbox/mtex#2181] [Ask Anything]
While using AZTEC Crystal to prepare my EBSD data for analysis in MTEX, I noticed that multiple grains had small (2-4 pixel) groups of differently oriented data (**can you give approximate misorientations?**). I want to remove these pixels because they are indexing as grains within MTEX. This picture shows the sample within AZTEC Crystal, with multiple grains with this problem (the two blue-green grains with the purple and red pixels in the top of the pictures for example):
![image](https://github.com/user-attachments/assets/b0b84e8a-2c85-47e2-bd5b-9f3fe7bf9dd5)
 
This picture shows measurements within the same blue-green grain from above:
![image](https://github.com/user-attachments/assets/9faa0ce2-6717-418b-a647-8e0e7893e157)

I have already tried the following:
 
**Within AZTEC**:
- Auto-Clean Up
- Remove Wild Spikes (removes single points but does not change points that have 1+ neighbors)
- Remove Pseudo-Symmetries to remove any twins (using the measure tool to measure misorientations and then append them to the tool, removed some of the misorientations but not all)

Sample before cleaning routine:
![image](https://github.com/user-attachments/assets/d0565895-45e2-45fa-a3d3-3475d268a244)
 
Sample after cleaning routine:
![image](https://github.com/user-attachments/assets/5398fa67-359d-4293-aded-388bed1e776a)

Although the sample is better after cleaning in MTEX, there are still 2-4 pixel pseudo grains within larger grains (examples circled above). Because of this, I decided to try to clean the sample using MTEX.
 
**Within MTEX** (used only a subregion of the sample for testing to save data processing time):
Here is the subregion after grain reconstruction:
![image](https://github.com/user-attachments/assets/9cc53ffe-5aec-4456-b3b2-98dc85a7ab14)

- Remove grains based on pixel size (up to pixel sizes of 50, this removed other grains but did not remove the ones I was looking at)
![image](https://github.com/user-attachments/assets/dddb1691-8972-40fb-9fe6-76b057110484)
- Apply the Mean, Median, Smoothing Spline, and Halfquadratic filters (both alone and in combination, following the workflow from the Denoising Orientation Maps page on the MTEX website)
- Used a quotient value and changed it to remove grains (removed all grains below 1, 1.5, and 2 to test; but the small pseudo grains remained)
![image](https://github.com/user-attachments/assets/8c9914d5-96b0-44cd-a140-ecde3222978e)

Thank you for any help you can provide!
Jake

**Comment by zmichels** [comment]
Would you be able/willing to post a link to some of the (uncleaned) data? I believe this is surmountable with an iterative grain construction approach/technique similar to what you have tried... but perhaps slightly different in detail (which may depend on the dataset).
